src/main/scala/org/opennetworkinsight/dns/DNSPostLDA.scala

```diff
@@ -142,8 +143,28 @@ object DNSPostLDA {
     addcol("score")
 
     logger.info("Persisting data")
-    val scored = src_scored.filter(elem => elem._1 < threshold).sortByKey().map(row => row._2.mkString(","))
-    scored.saveAsTextFile(resultsFilePath)
+
+
+    val filtered = src_scored.filter(elem => elem._1 < threshold)
+
+    val count = filtered.count
+
+    val takeCount  = if (topK == -1 || count < topK) {
+      count.toInt
+    } else {
+      topK
+    }
+    class OrderByLeastScoreToTop() extends Ordering[(Double,Array[Any])] {
+      def compare(p1: (Double, Array[Any]), p2: (Double, Array[Any]))    = p2._1.compare(p1._1)
+    }
+
+    implicit val ordering = new OrderByLeastScoreToTop()
+
+    val top : Array[(Double,Array[Any])] = filtered.top(takeCount)
+
+    val outputRDD = sc.parallelize(top).sortBy(_._1).map(_._2.mkString("\t"))
```

Comment:
WHOAH!!!

I almost checked in a major change that would have broken OA -- tab delimited output.

This change should be handled separately and synchronized with OA.

will revert this checkin to use comma delimited output